### PR TITLE
Conform to the PRISM guidelines (step 1/2)

### PR DESCRIPTION
**Pull request by quentinblampey**
- [x] use `ruff` and fix all linting / formatting issues
- [x] add CODEOWNERS file (@HakimBenkirane)
- [x] update CI for a PyPI release (⚠️ need to add the `PYPI_TOKEN` secret)
- [x] removed the (deprecated) argparse entrypoint - use the API instead
- [x] cleanup the imports (more imports in sub modules to avoid long imports, + use relative imports instead of absolute)
- [x] fix docstrings: duplicates type hints (provided in both type hints and docstrings)
- [x] configure the logger at the package level (with colors)
- [x] added a toy dataset loader so that we don't need to download the toy data manually
- [x] minor modules refactor
- [x] cleanup docs: use only one ` instead of two `` + fix broken links
- [x] shorter README (the rest will be in the docs) + add logo and badges
- [x] removed the `src` layout
- [x] `model.explain` in the tuto returns an error for me, is it normal @HakimBenkirane?
- [ ] (Optional) more input validation could be added to provide clear messages when the inputs are wrong (and how to debug)
- [ ] add a tuto about hyperparameter tuning (Hakim)
- [ ] avoid passing `label` and `event` variables to all functions (see one of my comments below)
- [ ] Re-update the readme usage guide to include the toy dataset (and check that the whole thing is running smoothly)
- [ ] Add more tests + add a CRON as in Sopa/Novae so that the tests are run multiple times every week
- [ ] Check that Hakim is okay with the logo
- [x] Fix (manually?) the tuto .md so that it renders cleaner

An additional PR will be needed after the first release to add the docs to the README.

### CODE REVIEW

**Review by HakimBenkirane — APPROVED** [review]
Changes correctly implemented and package behaviour seems fine. I am ok with this version
Still to do:
- Find better ways to represent the arguments
- PyPI token asap

**Comment by quentinblampey** [comment]
@HakimBenkirane I also thought about something else:
I saw that in most functions, we have to duplicate the same params: `label="PAM50", event="OS", surv_time="OS.time"`

I see two solutions to avoid that:
- (easy) the first time that we provided these arguments, we store them in `clinical_df.attrs`, and we recover them at each downstream call
- (medium) move from a `dict[str, DataFrame]` data structure to a `MuData` object (multimodal AnnData). Then, similar as above, we can store the arguments, but in `mdata.uns` probably makes more sense. Note also that one single `mdata` object will contain both `omics_df` and `clinical_df`, which seems much cleaner to me!

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <p align="center">
-  <img src="https://raw.githubusercontent.com/prism-oncology/customics/main/docs/assets/customics.png" alt="customics_logo" width="300"/>
+  <img src="./docs/assets/customics.png" alt="customics_logo" width="300"/>
```
Comment (quentinblampey) [inline README.md]: Please keep the absolute path here (`https://raw.githubusercontent.com/prism-oncology/customics/main/docs/assets/customics.png`) because otherwise it will not work for external websites, e.g., on pypi the page will show a "image not found" error since the relative path won't work

File: README.md [hunk 2]
```diff
@@ -27,90 +31,64 @@ It relies on [`mudata`](https://mudata.readthedocs.io/stable/), a core [scverse]
...
-    event="OS",          # survival event column (0/1)
-    surv_time="OS.time", # survival time column
-)
-model.fit(
-    mdata=mdata,
-    omics_val=omics_val, # optional validation set
-    batch_size=32,
-    n_epochs=30,
-    verbose=True,
+    device=torch.device("cuda" if torch.cuda.is_available() else "cpu"),
```
Comment (quentinblampey) [inline README.md]: Can we make `device` an optional argument?
I.e., in the model initialization, we can have `device: str | None = None`, and by default (i.e., if `device is None`), we use `device=torch.device("cuda" if torch.cuda.is_available() else "cpu")`

File: README.md [hunk 3]
```diff
@@ -27,90 +31,64 @@ It relies on [`mudata`](https://mudata.readthedocs.io/stable/), a core [scverse]
...
-ci = model.evaluate(
-    mdata,
-    task="survival",
-)
+metrics = model.evaluate(mdata, task="classification")  # Accuracy, F1, AUC, …
+ci = model.evaluate(mdata, task="survival")             # concordance index
 
 # --- 5. Visualise & explain ---
 model.plot_loss()
 model.plot_representation(mdata, label="PAM50",
```
Comment (quentinblampey) [inline README.md]: Is `label` needed here? Since we ran `prepare_input`?

File: README.md [hunk 4]
```diff
@@ -27,90 +31,64 @@ It relies on [`mudata`](https://mudata.readthedocs.io/stable/), a core [scverse]
...
                           filename="latent_space", title="t-SNE of latent space")
-model.stratify(omics_train, clinical_df, event="OS", surv_time="OS.time")
-model.explain(sample_ids, omics_train, clinical_df,
-              source="rna", subtype="Her2", label="PAM50")
+model.stratify(mdata, event="OS", surv_time="OS.time",
```
Comment (quentinblampey) [inline README.md]: Same here for `event="OS", surv_time="OS.time"`

File: README.md [hunk 5]
```diff
@@ -27,90 +31,64 @@ It relies on [`mudata`](https://mudata.readthedocs.io/stable/), a core [scverse]
...
+               save_path="results/km_stratification", show=True)
+model.explain(sample_ids, mdata, source="rna", subtype="Her2", label="PAM50")
```
Comment (quentinblampey) [inline README.md]: Same for `label="PAM50"` here

File: README.md [hunk 6]
```diff
@@ -27,90 +31,64 @@ It relies on [`mudata`](https://mudata.readthedocs.io/stable/), a core [scverse]
...
 
 # --- 5. Visualise & explain ---
 model.plot_loss()
 model.plot_representation(mdata, label="PAM50",
                           filename="latent_space", title="t-SNE of latent space")
-model.stratify(omics_train, clinical_df, event="OS", surv_time="OS.time")
-model.explain(sample_ids, omics_train, clinical_df,
-              source="rna", subtype="Her2", label="PAM50")
+model.stratify(mdata, event="OS", surv_time="OS.time",
+               save_path="results/km_stratification", show=True)
```
Comment (quentinblampey) [inline README.md]: I think we can remove `save_path="results/km_stratification"` in this README to make the example lighter

**Comment by alihamraoui** [comment]
> Some comments again @alihamraoui 😅

Thanks, Quentin, for the quick review!

File: README.md (hunk 3 above)
Comment (alihamraoui) [reply README.md]: It's the col in `mdata.obs` used for coloring (according to the docstring), so different concept than what `prepare_input` stores. I think we should rename it to `col_label`, what do you think?

Comment (quentinblampey) [reply README.md]: Ah okay I see, thanks for the clarification! We could probably use `color` then, for consistency with `scanpy`?
